fix: return 409 when pushing duplicate version without overwrite param

lanej opened the pull request:
Rubygems does not display errors when the response code is successful.  Previously you would only see the error message if you were doing a verbose push.  Without showing the raw HTTP logs a false positive is produced.

By changing the status code to 409 the error will be displayed to the actor.  This is not a functional change server-side.

lanej committed:
fix: return 409 when pushing duplicate version without overwrite param

Rubygems does not display errors when the response code is successful.
Previously you would only see the error message if you were doing a
verbose push.  It also gives the actor a false sense of success.

By changing the status code to 409 the error will be displayed to the
author.  This is not a functional change server-side.

drbrain commented:
Can you paste in a `rake test` run?

lanej commented:
@drbrain ugh no CI.  of course.

```
Run options: --seed 16793

# Running:

Errno::ENOENT:No such file or directory @ rb_file_s_stat - /tmp/geminabox-test-data/specs.4.8
/Users/jlane/p/geminabox/lib/geminabox/indexer.rb:24:in `stat'
/Users/jlane/p/geminabox/lib/geminabox/indexer.rb:24:in `updated_gemspecs'
/Users/jlane/p/geminabox/lib/geminabox/server.rb:54:in `reindex'
/Users/jlane/p/geminabox/lib/geminabox/gem_store.rb:82:in `write_and_index'
/Users/jlane/p/geminabox/lib/geminabox/gem_store.rb:20:in `save'
/Users/jlane/p/geminabox/test/units/geminabox/gem_store_test.rb:48:in `test_ensure_gem_unique'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:108:in `block (3 levels) in run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:205:in `capture_exceptions'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:105:in `block (2 levels) in run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:256:in `time_it'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:104:in `block in run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:331:in `on_signal'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:276:in `with_info_handler'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest/test.rb:103:in `run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:778:in `run_one_method'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:305:in `run_one_method'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:293:in `block (2 levels) in run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:292:in `each'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:292:in `block in run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:331:in `on_signal'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:318:in `with_info_handler'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:291:in `run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:152:in `block in __run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:152:in `map'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:152:in `__run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:129:in `run'
/Users/jlane/.gem/ruby/2.3.1/gems/minitest-5.8.4/lib/minitest.rb:56:in `block in autorun'
Generating Marshal quick index gemspecs for 1 gems
.
Complete
Generated Marshal quick index gemspecs: 0.001s
Generating specs index
Generated specs index: 0.000s
Generating latest specs index
Generated latest specs index: 0.000s
Generating prerelease specs index
Generated prerelease specs index: 0.000s
Compressing indices
Compressed indices: 0.001s
................................................................................

Finished in 0.147108s, 543.8192 runs/s, 863.3130 assertions/s.

80 runs, 127 assertions, 0 failures, 0 errors, 0 skips
WARN: tilt autoloading 'tilt/erb' in a non thread-safe way; explicit require 'tilt/erb' suggested.
Run options: --seed 64986

# Running:

..Generating Marshal quick index gemspecs for 1 gems
.
Complete
Generated Marshal quick index gemspecs: 0.001s
Updated indexes: 0.001s
Compressing indices
Compressed indices: 0.001s
.Generating Marshal quick index gemspecs for 1 gems
.
Complete
Generated Marshal quick index gemspecs: 0.001s
Updated indexes: 0.001s
Compressing indices
Compressed indices: 0.001s
...

Finished in 0.475659s, 12.6141 runs/s, 25.2281 assertions/s.

6 runs, 12 assertions, 0 failures, 0 errors, 0 skips
Run options: --seed 2456

# Running:

......................

Finished in 12.338360s, 1.7831 runs/s, 4.2955 assertions/s.

22 runs, 53 assertions, 0 failures, 0 errors, 0 skips
```

drbrain commented:
My name is Eric Hodel and I approve this pull request.

Not sure what to do about the shapely, though.

lanej commented:
@drbrain had a `.travis.yml` file already. just required activation on https://travis-ci.org/fastly/geminabox

lanej committed:
chore: set CI rvm version to 2.3.0

Fixes public_suffix > 2.0 not being available on ruby < 2.0.

Our deployment of geminabox is currently running exclusively in 2.3.0.

See:
https://github.com/fastly-cookbooks/gem-server/blob/fe976d22715748b9f5635f886e0a648160d18723/recipes/default.rb#L11

lanej committed:
test: fix test setup for http client tests

Context of the follow few test changes is that this repository is
getting moldy and some updated dependencies are causing unexpected
behavior.

Fix stub matching by manually creating the base 64 encoded Authorization
header.  This test passed before I ran `bundle update` and failed
immediately on CI (which always has the latest versions of the
dependencies).

lanej committed:
test: remove duplicate tests

A few tests had the same method name meaning the last definition was the
only one that is valid.
